action/protocol/staking/bucket_pool.go

```diff
@@ -161,11 +151,16 @@ func (bp *BucketPool) Clone() *BucketPool {
 
 // SyncPool sync the data from state manager
 func (bp *BucketPool) SyncPool(sm protocol.StateManager) error {
-	if bp.exist {
-		_, err := sm.State(bp.total, protocol.NamespaceOption(StakingNameSpace), protocol.KeyOption(bucketPoolAddrKey))
-		return err
+	if bp.enableSMStorage {
+		switch _, err := sm.State(bp.total, protocol.NamespaceOption(StakingNameSpace), protocol.KeyOption(bucketPoolAddrKey)); errors.Cause(err) {
+		case nil:
+			return nil
+		case state.ErrStateNotExist:
```

Comment:
this might be an actual error?
say if value is stored at Greenland+1, now it comes in at Greenland+5 and sees StateNotExist